Improve formatting of e2e test build errors

I was editing `testcases/key-value/lib.rs` and missed a `?` on a fallible operation and this is how `make test-kv` explained the problem to me:

![image](https://user-images.githubusercontent.com/865538/236047502-ce1d5dc3-71f5-4d83-9997-49e2cf3c3d48.png)

It would be great if we could preserve line breaks to make errors like this easier to read.

Hi @itowlson 

sorry about this. I am looking into this.

would something like following be fine:

```
running 1 test
stderr: 
[
    "   Compiling key-value v0.1.0 (/Users/rajatjindal/go/src/github.com/fermyon/spin/tests/testcases/key-value)",
    "error[E0277]: can't compare `[_]` with `Result<Vec<u8>, key_value::key_value::Error>`",
    "  --> src/lib.rs:26:28",
    "   |",
    "26 |     ensure!(b\"baz\" as &[_] == &store.get(\"bar\"));",
    "   |                            ^^ no implementation for `[_] == Result<Vec<u8>, key_value::key_value::Error>`",
    "   |",
    "   = help: the trait `PartialEq<Result<Vec<u8>, key_value::key_value::Error>>` is not implemented for `[_]`",
    "   = help: the following other types implement trait `PartialEq<Rhs>`:",
    "             <&[B] as PartialEq<[A; N]>>",
    "             <&[T] as PartialEq<Vec<U, A>>>",
    "             <&[u8] as PartialEq<BytesMut>>",
    "             <&[u8] as PartialEq<bytes::Bytes>>",
    "             <&mut [B] as PartialEq<[A; N]>>",
    "             <&mut [T] as PartialEq<Vec<U, A>>>",
    "             <[A; N] as PartialEq<&[B]>>",
    "             <[A; N] as PartialEq<&mut [B]>>",
    "           and 8 others",
    "   = note: required for `&[_]` to implement `PartialEq<&Result<Vec<u8>, key_value::key_value::Error>>`",
    "   = note: 1 redundant requirement hidden",
    "   = note: required for `&&[_]` to implement `~const PartialEq<&&Result<Vec<u8>, key_value::key_value::Error>>`",
    "",
    "For more information about this error, try `rustc --explain E0277`.",
    "error: could not compile `key-value` (lib) due to previous error",
    "Error: Build command for component hello failed with status Exited(101)",
]
stdout: 
[
    "Executing the build command for component hello: cargo build --release --target wasm32-wasi",
]
```

I am getting above using

```
println!(
            "stderr: \n{:#?}",
            std::str::from_utf8(&output.stderr)?
                .lines()
                .collect::<Vec<&str>>()
        );
```

If there is a better way to do this, could you please let me know. thanks

--- a/crates/e2e-testing/src/spin_controller.rs
+++ b/crates/e2e-testing/src/spin_controller.rs
@@ -77,7 +77,7 @@ impl Controller for SpinUp {
                         utils::get_output_from_stderr(Some(stderr_stream), Duration::from_secs(2))
                             .await?;
                     return Err(anyhow!(
-                        "error running spin up.\nstdout {:?}\nstderr: {:?}\n",
+                        "error running spin up.\nstdout {:#?}\nstderr: {:#?}\n",
                         stdout_logs,
                         stderr_logs
                     ));

--- a/crates/e2e-testing/src/utils.rs
+++ b/crates/e2e-testing/src/utils.rs
@@ -45,9 +45,16 @@ pub fn run<S: Into<String> + AsRef<OsStr>>(
     let output = cmd.output()?;
     let code = output.status.code().expect("should have status code");
     if code != 0 {
-        println!("{:#?}", std::str::from_utf8(&output.stderr)?);
-        println!("{:#?}", std::str::from_utf8(&output.stdout)?);
-        panic!("command `{:?}` exited with code {}", cmd, code);
+        let stdout = std::str::from_utf8(&output.stdout)?
+            .lines()
+            .collect::<Vec<&str>>();
+        let stderr = std::str::from_utf8(&output.stderr)?
+            .lines()
+            .collect::<Vec<&str>>();
+        panic!(
+            "command `{:?}` exited with code {}.\n\nstdout:\n{:#?}\n\nstderr:\n{:#?}\n",
+            cmd, code, stdout, stderr
+        );
     }
 
     Ok(output)

